### PR TITLE
Feature/voting table filters

### PR DESCRIPTION
**Pull request by timganter**
Some progress towards https://github.com/CouncilDataProject/seattle/issues/18. Feel free to edit/change/modify whatever. 

1. Some checkboxes to filter table by type of vote cast.
1. Some styling tweaks to use https://react.semantic-ui.com instead of custom styling.

### Some things to be aware of.

1. @tohuynh mentioned that there is a limit to how many records are fetched at once. This means that if the number of rows in the table is beyond the limit, then there could be rows not fetched that the filter is missing. Which would be the case anyway I suppose. That the user wouldn't see all the records to begin with at the moment.
1. Currently there is no refetching of results. The filtering is all done on the front end without any new call to the server. This might all need to be revised someday, but this PR has a working version of the `Vote` column in the table.
1. You'll notice some stickyness when they're all checked then unchecked or when `In Favor` is checked. Because `In Favor` typically has the majority of rows. Adding some paging or infinite scrolling could help remove that stickyness from the re-rendering.

### CODE REVIEW

**Review by evamaxfield — COMMENTED** [review]
Thanks for the PR! Looks great! My only request is if you could enforce that none of the columns change width on filter?

cdp-vote-column-size-change

**Comment by evamaxfield** [comment]
There is also some weirdness when there are more check boxes?
image

**Comment by evamaxfield** [comment]
> There is also some weirdness when there are more check boxes?

Correction.... The vote has no value.... Interesting. So not a bug on your side but could you include a check for when the vote value is missing?

**Comment by timganter** [comment]
Nice catch! https://github.com/CouncilDataProject/seattle/pull/40/commits/458832d2038e7f32629e1bf60c1bd58187be2688 and https://github.com/CouncilDataProject/seattle/pull/40/commits/1876443c463077aceeb1849f6087d8e902e2e154 should fix those. 

If all else looks good. Maybe we can merge and make new issues for anything else you find. 

Adding that no-wrap thing, I noticed that https://github.com/timganter/seattle/blob/feature/votingTableFilters/src/components/VotingTable.jsx could use some refactoring/tweaking to use react semantic ui and add that `singleLine={true}` to the Matter, Decision, and Name columns. 

That could be an easy issue for someone at the next hack-a-thon to tackle.

File: src/components/Person.jsx [hunk 1]
```diff
@@ -1,32 +1,66 @@
...
-                <a  target={'_blank'}
-                    rel="noopener noreferrer"
-                    href={website}>{website}</a>
-            </div>
-        </Card>
-        <VotingTable votingData={votes} isPerson={true} />
-    </FlexContainer>
-    )
+  // == Get unique vote cast types from votes list.
+  const votingOptions = _.compact(_.uniq(votes.map(vote => vote.voteForPerson)))
```
Comment (evamaxfield) [inline src/components/Person.jsx:10]: Saw this `_.compact` added in a commit. Nice. Thanks lodash :tada:

**Review by evamaxfield — APPROVED** [review]
Thanks for the quick updates. This looks great! Will merge and make issues for improving the responsiveness of the table. I think with the comment on refactoring/ tweaking of using react semantic ui but also about pagination on PR description I think this table will need to be refactored a bit to make database requests given a set of filters for sure. I don't _think_ that should slow it down too much if we optimize the database.

**Pull request merged**